asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -470,6 +470,32 @@ def get_config(self, context, id):
 
         return dict(result)
 
+    @registry.receives(resources.ROUTER_GATEWAY, [events.BEFORE_CREATE])
+    @log_helpers.log_method_call
+    def ensure_bdvif_limit(self, resource: str, event: str, trigger, payload: events.DBEventPayload):
+        """
+        There is a hardware limit on ASR1K/C8500 that allows us to not bind more than X BD-VIFs in a BD.
+        As configuration would just silently fail, we try to catch the situation on API level.
+        """
+        context = payload.context
+        network_id = payload.metadata['network_id']
+        router_id = payload.resource_id
+
+        host = self.get_l3_agents_hosting_routers(context, [router_id])
+        if len(host) == 0:
+            host = None
+        else:
+            host = host[0].host
+
+        port_count = self.db.get_port_count_in_segment(context, network_id)
+
+        # Either the router is scheduled and the agent has no room for BD-VIFs in that BD
+        # or it is not scheduled and all existing agents are above the limit
+        limit = cfg.CONF.asr1k_l2.bdvif_bd_limit
+        if (host and port_count[host] >= limit) \
```

Comment:
I think 100 is stil ok. Why do you mean I keep 1 as a spare. It rejects the interface assignment if there are already 100 or more ports bound in that segment.

Reply:
As you said, it rejects the assignment for 100 or more. If 100 is okay we could still allow 100, but the code is written so that 100 is not allowed. Just wanted to know if this is intentional or an off by one - or if I misunderstood something again.